File: presentation/src/main/java/io/tripled/social/client/presentation/request/WallCliRequest.java
package io.tripled.social.client.presentation.request;

import io.tripled.social.client.presentation.CliRequest;
import io.tripled.social.client.presentation.Output;

public class WallCliRequest implements CliRequest {
  private String userName;

  public WallCliRequest(String userName) {
    this.userName = userName;
  }

  @Override
  public void accept(Visitor visitor, Output output) {
    visitor.visit(this, output);
  }

  public String getUserName() {
    return userName;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;

    WallCliRequest request = (WallCliRequest) o;

    return userName.equals(request.userName);
  }

  @Override
  public int hashCode() {
    return userName.hashCode();
  }
}

File: src/main/kotlin/io/tripled/social/client/infrastructure/cli/request/WallCliRequest.kt
package io.tripled.social.client.infrastructure.cli.request

import io.tripled.social.client.infrastructure.cli.CliRequest
import io.tripled.social.client.infrastructure.cli.Output

class WallCliRequest(val userName: String) : CliRequest {
    override fun accept(visitor: Visitor, output: Output) {
        visitor.visit(this, output)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false

        val request = other as WallCliRequest

        return userName == request.userName
    }

    override fun hashCode(): Int {
        return userName.hashCode()
    }
}
